### DISCUSSION TITLE
How to correct this grammar?

### DISCUSSION THREAD

**Discussion by panlina** [tree-sitter/tree-sitter#777] [Q&A]
I have this:

```
module.exports = grammar({
	name: 'a',
	rules: {
		expression: $ => choice(
			$.number,
			$.expression_call,
			$.expression_unary,
			$.expression_binary
		),
		expression_binary: $ =>
			prec.left(0, seq(
				field('left', $.expression),
				field('operator', '+'),
				field('right', $.expression)
			)),
		expression_unary: $ =>
			prec(1, seq(
				field('operator', '+'),
				field('right', $.expression)
			)),
		expression_call: $ => prec.right(2,
			seq(
				$.expression,
				$.expression
			)
		),
		number: $ => /\d+/,
	},
	conflicts: $ => [
		[$.expression_binary, $.expression_unary, $.expression_call]
	]
});
```

Basically it's ordinary unary and binary expressions, with function call which is direct concatenation of callee and argument.

I want to parse `+1+2 3+4` as `(+1)+(2 3)+4`, but what I get is `((+1) (+(2 3))) (+4)`.

Have been fiddled around this for a while but didn't get a difference. I don't know about LR parser knowledges. Can anyone give some help? Thanks.

Ignore the weirdness that I let function call apply a number to another number. It's just a simplified example.

**Comment by maxbrunsfeld** [comment]
Can you post this question to the [discussion board](https://github.com/tree-sitter/tree-sitter/discussions) instead? We would like to use the issues list more for tracking work that we need to do on Tree-sitter itself.

**Reply by maxbrunsfeld** [reply]
@panlina I realized that I can automatically convert an issue to a discussion. I've done that for this thread.

**Reply by panlina** [reply]
Thanks, Max.
